fyrox-core-derive/src/inspect/utils/prop_keys.rs

```diff
@@ -6,12 +6,36 @@ use proc_macro2::TokenStream as TokenStream2;
 use quote::quote;
 use syn::*;
 
-use crate::inspect::{args, utils};
+use crate::inspect::args;
 
-/// Returns a list of `pub const [VARIANT_]FIELD: &'static str = "key_value"`;
+/// `pub const [VARIANT_]FIELD: &'static str = "key";`
+pub fn prop_keys_impl(ty_args: &args::TypeArgs) -> TokenStream2 {
+    let ty_ident = &ty_args.ident;
+    let (impl_generics, ty_generics, where_clause) = ty_args.generics.split_for_impl();
+
+    let prop_keys = self::quote_prop_keys(ty_args);
+
+    quote! {
+        /// Property key constants
+        impl #impl_generics #ty_ident #ty_generics #where_clause {
+            #prop_keys
+        }
+    }
+}
+
+/// List of `pub const <ident> = <name>;`
+///
+/// # Definition format
+///
+/// | Type         | Identifier                | Name                     |
+/// |--------------|---------------------------|--------------------------|
+/// | Struct       | `FIELD_NAME`              | `field_name`             |
+/// | Unit struct  | `F_<number>`              | `<number>`               |
+/// | Tuple struct | `VARIANT_NAME_FIELD_NAME  | `VariantName.field_name` |
```

Comment:
Looks like `Tuple struct` should be swapped with `Enum`, because otherwise it looks odd to have `VARIANT_NAME_F_<number>` for enumerations 🤔

Reply:
Right! I've updated it to this:

```rust
/// | Struct         | `FIELD_NAME`              | `field_name`             |
/// | Struct (tuple) | `F_<number>`              | `<number>`               |
/// | Enum (struct)  | `VARIANT_NAME_FIELD_NAME  | `VariantName.field_name` |
/// | Enum (tuple)   | `VARIANT_NAME_F_<number>` | `VariantName.<number>`   |
```